kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportNamer.kt

```diff
@@ -902,6 +922,45 @@ private fun ObjCExportMapper.canHaveSameName(first: PropertyDescriptor, second:
     return bridgePropertyType(first) == bridgePropertyType(second)
 }
 
+private fun DeclarationDescriptor.getObjCName(forSwift: Boolean): String {
```

Comment:
Ideally yes. But IMO it should report a warning for all mangled names (even if the annotation isn't present).

Reply:
Yes, I agree. So, in case of a clash, it probably makes sense to emit
- an error, if the mangling is applied to something with `@ObjCName`
- a warning, otherwise

What do you think?

Generally, current implementation of naming in ObjCExport makes this not simple, so here I'm mostly trying to bring us at the same page regarding the concept, not insisting on implementing this as a part of this PR.

Reply:
Yeah I like that! Specifying conflicting `@ObjCName` annotations is definitely an error.
But that would indeed be a big addition for this PR, so probably best to add that separately.